spec/account/account-manager.spec.js

```diff
@@ -0,0 +1,154 @@
+"use strict";
+
+describe("AccountManager", () => {
+    var accountManager, otDateMock;
+
+    beforeEach(() => {
+        var AccountManager, config, promiseMock, randomMock;
+
+        AccountManager = require("../../lib/account/account-manager");
+        otDateMock = require("../mock/ot-date-mock");
+        promiseMock = require("../mock/promise-mock");
+        randomMock = require("../mock/random-mock");
+        class AccountServiceFake {
+            constructor() {
+                this.initiate = jasmine.createSpy("initiate");
+                this.initiate.andCallFake((accountId, accountInfo, options) => {
+                    return promiseMock.resolve({
+                        accountId: accountInfo.accountId,
+                        email: accountInfo.email,
+                        mfa: accountInfo.mfa,
+                        salt: accountInfo.salt
+                    });
+                });
+                this.complete = jasmine.createSpy("complete");
+                this.complete.andCallFake(() => {
+                    return promiseMock.resolve(true);
+                });
+                this.get = jasmine.createSpy("get");
+                this.get.andCallFake(() => {
+                    return promiseMock.resolve({
+                        mfa: "339r93939303093"
+                    });
+                });
+            }
+        }
+        class HotpFake {
+            constructor() {
+                this.generateSecret = jasmine.createSpy("generateSecret");
+                this.generateSecret.andCallFake(() => {
+                    return promiseMock.resolve("thisisasecrectcodefrommfa");
+                });
+                // This is a sync call
+                this.verifyToken = jasmine.createSpy();
+                this.verifyToken.andCallFake((key, token, opts) => {
+                    if (token === "987654") {
+                        return false;
+                    }
+
+                    return true;
+                });
+            }
+        }
+        config = {
+            account: {
+                completeLifetime: {
+                    months: 6
+                },
+                initiateLifetime: {
+                    hours: 1
+                }
+            },
+            hotp: {
+                previous: {
+                    afterDrift: 1,
+                    beforeDrift: 1,
+                    drift: 2
+                }
+            }
+        };
+        accountManager = new AccountManager(new AccountServiceFake, config, new HotpFake, otDateMock, randomMock, promiseMock);
```

Comment:
accountManager = new AccountManager(new AccountServiceFake(), config, new HotpFake(), otDateMock, randomMock, prom....

You should use () whenever you use new.